dev/doc/build-system.dev.txt

```diff
@@ -87,7 +87,7 @@ Cons:
    clear-text generated .ml.
 
 
-Makefiles hierachy
+Makefiles hierarchy
 ------------------
```

Comment:
I think this might cause warnings (the underline will be too short)

Reply:
Not in a .txt surely

Reply:
Oh, sorry, I thought this was an rst